src/net.c

```diff
@@ -2686,16 +2715,8 @@ handle_request(coap_context_t *context, coap_session_t *session, coap_pdu_t *pdu
           response->used_size = 0;
         }
 
-        if ((respond == RESPONSE_SEND)
```

Comment:
This change drops suppression of multicast error responses.

Reply:
That is what I initially thought.  However, 
````
#ifndef coap_mcast_interface
# define coap_mcast_interface(Local) 0
#endif
````
means that it is a no-op.  However, `coap_send()` calls `coap_send_pdu()` which has
````
  /* Do not send error responses for requests that were received via
  * IP multicast.
  * FIXME: If No-Response option indicates interest, these responses
  *        must not be dropped. */
  if (coap_is_mcast(&session->addr_info.local) &&
    COAP_RESPONSE_CLASS(pdu->code) > 2) {
    return COAP_DROPPED_RESPONSE;
  }
````
and agree the FIXME is relevant for RFC7967

> The server MUST send back responses of the classes for which the
> client has not expressed any disinterest.

I will look at moving the `coap_send_pdu() `code into `no_respond()` to keep things in one place.

Reply:
Oh, ok, I have missed that part. Moving into no_response is a good idea, anyway.

Reply:
Code change for mcast done and code pushed.